src/relax/op/tensor/binary.h

```diff
@@ -25,18 +25,38 @@
 #ifndef TVM_RELAX_OP_TENSOR_BINARY_H_
 #define TVM_RELAX_OP_TENSOR_BINARY_H_
 
-#include <tvm/relax/expr.h>
-#include <tvm/relax/type.h>
-
-#include <algorithm>
-#include <vector>
-
 #include "../op_common.h"
 
 namespace tvm {
 namespace relax {
 
-StructInfo InferStructInfoBroadcast(const Call& call, const BlockBuilder& ctx);
+/*! Quick helper macro
+ * - Expose a positional make function to construct the node.
+ * - Register op to the registry.
+ *
+ * We make the decision to always only expose positional argument.
+ * We will do rewrapping in the frontend to support language
+ * sugars such as keyword arguments and default value.
+ *
+ * \param OpName the name of registry.
+ */
+#define RELAX_REGISTER_BINARY_OP(OpName)                                          \
+  TVM_REGISTER_GLOBAL("relax.op." OpName).set_body_typed([](Expr lhs, Expr rhs) { \
+    static const Op& op = Op::Get("relax." OpName);                               \
+    return Call(op, {lhs, rhs}, Attrs(), {});                                     \
+  });                                                                             \
+  RELAX_REGISTER_OP("relax." OpName)                                              \
+      .set_num_inputs(2)                                                          \
+      .add_argument("lhs", "Tensor", "The left hand side tensor.")                \
+      .add_argument("rhs", "Tensor", "The right hand side tensor.")
+
+#define RELAX_REGISTER_BINARY_BROADCAST_OP(OpName)                                \
+  RELAX_REGISTER_BINARY_OP(OpName).set_attr<FInferStructInfo>("FInferStructInfo", \
+                                                              InferStructInfoBroadcast)
+
+#define RELAX_REGISTER_CMP_OP(OpName)                                             \
+  RELAX_REGISTER_BINARY_OP(OpName).set_attr<FInferStructInfo>("FInferStructInfo", \
+                                                              InferStructInfoCMP)
```

Comment:
It seems to me that we can move the macro definitions to .cc side. I’m not sure if we would need to go with this header file.

Reply:
Decided to localize the macros to .cc files.